### PR TITLE
Ormolu in default.nix and emacs before-save hook.

### PR DESCRIPTION
**Pull request by jacereda**

### CODE REVIEW

File: .dir-locals.el [hunk 1]
```diff
@@ -0,0 +1 @@
+((haskell-mode . ((before-save-hook . (ormolu-format-buffer)))))
```
Comment (eriksvedang) [inline .dir-locals.el:1]: This gives me a warning every time I visit a file in emacs.

Comment (eriksvedang) [reply .dir-locals.el:1]: (I already have my own save hook setup for Ormolu)

Comment (jacereda) [reply .dir-locals.el:1]: Right. How did you setup yours? It should be something local to the Carp directory (we don't want to format every codebase under the sun with ormolu).

Comment (eriksvedang) [reply .dir-locals.el:1]: True, I did set it up globally (which is not great).
Maybe there's a way to avoid the warning?

Comment (jacereda) [reply .dir-locals.el:1]: You can add it to the list of safe variables, but that's unsafe, somebody could add malicious code in a `.dir-locals.el` as a `before-save-hook`. There must be something better.

Comment (jacereda) [reply .dir-locals.el]: Can you try now? 

Using `0` in .dir-locals should allow disabling it for a directory.

**Comment by eriksvedang** [comment]
Worked well!

**Pull request merged**